### PR TITLE
Collect events

### PR DESCRIPTION
**Pull request by Elchi3**
Unfortunately, I see no more context for https://github.com/openwebdocs/mdn-bcd-collector/commit/8f23515e8d203b878178bb8f9bc874f3235bfccf. There was no pull request...

I'm suggesting to add event collections back but ignore everything that has to do with GlobalEventHandlers and WindowEventHandlers for now.

(Also, I know Serial events can be ignored due to bubbling to SerialPort. Added that exception in as well.)

### CODE REVIEW

File: test-builder/api.ts [hunk 1]
```diff
@@ -441,30 +449,27 @@ const buildIDLMemberTests = async (
...
     let expr: string | RawTestCodeExpr | RawTestCodeExpr[] = "";
 
-    // Constructors, constants, and static attributes should not have
+    // Constructors, constants, event handlers and static attributes should not have
     // auto-generated custom tests
     const customTestExactMatchNeeded =
-      isStatic || ["toString", "toJSON"].includes(member.name as string);
+      isStatic ||
+      isEventHandler ||
+      ["toString", "toJSON"].includes(member.name as string);
```
Comment (queengooborg) [inline test-builder/api.ts:472]: I think that auto-generating event handler tests from custom tests is actually what we want, as this ensures that we're using an instance when needed.

**Review by queengooborg — APPROVED** [review]

File: test-builder/api.ts (hunk 1 above)
Comment (Elchi3) [reply test-builder/api.ts]: If I leave this in, the generated test is what we want, I think:

```
"XMLHttpRequestEventTarget" in self && "onabort" in XMLHttpRequestEventTarget.prototype
```


If revert this, I the generated test is:

```
(function () {
  if (!("XMLHttpRequest" in self)) {
    return { result: false, message: "XMLHttpRequest is not defined" };
  }
  var instance = new XMLHttpRequest();
  return !!instance && "abort_event" in instance;
})();
```

`"abort_event" in instance` doesn't work of course, so I thought I needed to add `isEventHandler` to `customTestExactMatchNeeded`.

Thoughts?

**Comment by queengooborg** [comment]
> Unfortunately, I see no more context for [8f23515](https://github.com/openwebdocs/mdn-bcd-collector/commit/8f23515e8d203b878178bb8f9bc874f3235bfccf). There was no pull request...

This change was discussed during the BCD call on November 30, 2022.  The meeting notes state why they were disabled:

```
(Vinyl) Should we add event features for events that bubble up to interface, but interface has event handler properties? ([#18287](https://github.com/mdn/browser-compat-data/pull/18287), [#18288](https://github.com/mdn/browser-compat-data/pull/18288))
Philip: we removed a bunch of things re event handlers that were not needed
Philip: Some events that do bubble up were removed. Collector keeps adding entries back
Philip: BCD is now fairly consistent in that events show up where they happen. 
Vinyl: makes sense, no strong opinion either way. 
Philip: If we avoid having the collector add those events back, would that solve the problem?
Vinyl: Yes, seems that way. I can make changes so it’s not recreated on particular interfaces. Make collector disregard events in IDLs completely. Can handle things as custom implementations as needed
```

File: test-builder/api.ts (hunk 1 above)
Comment (Elchi3) [reply test-builder/api.ts]: ahh thank you! :)

**Comment by Elchi3** [comment]
I know the Global/WindowEventHandlers need a mapping before we can enable these again. We definitely don't want the collector to add these entries all over the place and that's why I skipped them in this PR. 

I also added Serial events to be ignored but there are surely more events that bubble which we want to ignore. I think we should probably have better ignore code than what I did for Serial here.

I'm thinking though that enabling this back on should allow us to test support for legit events again and discover new legit events? Seems like that is totally turned off now, right?

**Comment by Elchi3** [comment]
Main: 4657/15217 (30.60%) entries missing from collector that are in bcd
This branch: 4448/15217 (29.23%) entries missing from collector that are in bcd

**Review by queengooborg — APPROVED** [review]
Okay, I see that this doesn't add any new features either (by running `npm run find-missing-features bcd-from-collector`), so let's merge this.

**Pull request merged**